src/lib/optimizer/abstract_syntax_tree/union_node.cpp

```diff
@@ -49,23 +56,25 @@ std::shared_ptr<TableStatistics> UnionNode::derive_statistics_from(
 
 std::optional<ColumnID> UnionNode::find_column_id_by_named_column_reference(
     const NamedColumnReference& named_column_reference) const {
-  /**
-   * This function has to be overwritten if columns or their order are in any way redefined by this Node.
-   * Examples include Projections, Aggregates, and Joins.
-   */
   auto named_column_reference_without_local_alias = _resolve_local_alias(named_column_reference);
+
   if (!named_column_reference_without_local_alias) {
     return {};
-  } else {
-    const auto column_id_in_left =
-        left_child()->find_column_id_by_named_column_reference(*named_column_reference_without_local_alias);
+  }
+
+  // UnionNode can't resolve NamedColumnReference with table_name
+  if (named_column_reference_without_local_alias->table_name) {
```

Comment:
Would you say that asking a `UnionNode` for a named colref with a table name is a logical error? In that case, maybe even a `DebugAssert` could be placed here.

Reply:
it's not a logical error, might happen, another subtree might contain the NamedColumnRef.